### PR TITLE
Adds 'translateLabel()' method to HasLabel concerns

### PR DESCRIPTION
**Pull request by tiagof**
Hi.

I find myself using ->label() in a lot o places to simply translate the name of a component.
This is simply a UX nicety to have cleaner code.

Adds a new method `translateLabel(bool)`  to the several HasLabel concerns:
Form components, Actions, Columns and Filters

This allows the developer to replace something like:

```php
TextColumn::make('name')->label(__('name'))
```

with

```php
TextColumn::make('name')->translateLabel()
```

### CODE REVIEW

**Comment by danharrin** [comment]
Hi, what is the benefit of this?

**Comment by tiagof** [comment]
Hi, as mentioned in the op, it is a simple low impact change (imo) to provide a way to have cleaner code.
In my particular example I'd like to replace all of the `->label(..)` with this new method.
It prevents repeating strings...

```php
return $form
            ->schema([
                Forms\Components\TextInput::make('title')
                    ->label(__('Title'))
                   ...,
                Forms\Components\RichEditor::make('description')
                    ->label(__('Description'))
                    ...,
                Forms\Components\Grid::make()
                    ->columnSpan('full')
                    ->schema([
                        Forms\Components\DateTimePicker::make('start')
                            ->label(__('Start'))
                            ...,
                        Forms\Components\DateTimePicker::make('end')
                            ->label(__('End'))
                             ....
```

```php
return $form
            ->schema([
                Forms\Components\TextInput::make('title')
                    ->translateLabel()
                   ...,
                Forms\Components\RichEditor::make('description')
                    ->translateLabel()
                    ...,
                Forms\Components\Grid::make()
                    ->columnSpan('full')
                    ->schema([
                        Forms\Components\DateTimePicker::make('start')
                            ->translateLabel()
                            -...,
                        Forms\Components\DateTimePicker::make('end')
                            ->translateLabel()
                             ....
```

**Comment by danharrin** [comment]
My problem with this, is that it's generally bad practice to use translations with the English name as the key (like `__('Description')`), instead of nested keys like `__('filament.resources.post.description.label')`. Not sure if I want to be encouraging our users to do that?

You'll be pleased to know that theres an easy way to turn this on for all fields at once in a service provider, if this is merged:

```php
Forms\Components\Component::configureUsing(fn (Component $component) => $component-> translateLabel())
```

**Comment by tiagof** [comment]
I am **VERY** please to know there is a way to apply this (and others) globally 😁 - didn't know that!

Regarding the best practice note, imo, sometimes you just want to use the same translation everywhere, especially in common words like 'description'.
In more complex situations, the dev can/should continue to use `__('filament.resources.post.description.label')` for this.

File: packages/tables/src/Filters/Concerns/HasLabel.php [hunk 1]
```diff
@@ -9,19 +9,32 @@ trait HasLabel
...
     public function getLabel(): string
     {
-        return $this->evaluate($this->label) ?? (string) Str::of($this->getName())
+        $label = $this->evaluate($this->label) ?? (string) Str::of($this->getName())
             ->before('.')
             ->kebab()
             ->replace(['-', '_'], ' ')
             ->ucfirst();
+
+        return $this->shouldTranslateLabel
```
Comment (danharrin) [inline packages/tables/src/Filters/Concerns/HasLabel.php:36]: Please move this conditional onto one line, then we are ready to merge

**Review by danharrin — CHANGES_REQUESTED** [review]

File: packages/tables/src/Filters/Concerns/HasLabel.php (hunk 1 above)
Comment (tiagof) [reply packages/tables/src/Filters/Concerns/HasLabel.php]: done

**Comment by tiagof** [comment]
I'll update the docs and then "publish" the PR...

File: packages/tables/src/Filters/Concerns/HasLabel.php (hunk 1 above)
Comment (danharrin) [reply packages/tables/src/Filters/Concerns/HasLabel.php]: There are still places this needs to be fixed please

Comment (tiagof) [reply packages/tables/src/Filters/Concerns/HasLabel.php]: those have a complex if condition so I usually split lines in those cases.
Sure you want them in a single line?

**Comment by tiagof** [comment]
> I'll update the docs and then "publish" the PR...

done. Please check if the style/conventions are ok.
Thx

File: packages/tables/src/Filters/Concerns/HasLabel.php (hunk 1 above)
Comment (danharrin) [reply packages/tables/src/Filters/Concerns/HasLabel.php]: Fair enough, ok

**Comment by danharrin** [comment]
Thanks

**Pull request merged**